### PR TITLE
Update links in metadata section on single result page

### LINKED ISSUE #2731: Update Source and Provider links in the single result page Information section
## Problem

The links to Source and Provider on the single result page do not have an indication that they are external and open in a new tab. Moreover, they are inconsistent across the media types and link to the media foreign landing URL instead of the source or the provider (#560)

## Description

The links to creator in the [image](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/pages/image/_id/index.vue#L62-L73) and [audio](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioTrack/layouts/VFullLayout.vue#L32-L41) single result pages Information section should have an “external link” icon. Note: the `VLink` with an external link needs update because it doesn't wrap, and simply adding `show-external-link` to `VLink` creates single-line links, and long source names overlap other text on the page.

The audio creator link should also be updated to match the image creator link. It should be a conditional component: `VLink` if the `creator_url` is truthy and span if the `creator_url` is falsy.

Currently, the foreign_landing_url is linked to the “source” in the [image](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VImageDetails/VImageDetails.vue#L26-L28) page and “provider” in the [audio page](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioDetails/VAudioDetails.vue#L61C1-L63). The audio page should be updated to match the image page: the foreign_landing_url link should be added to the “source”, not provider.

## Additional context

It is possible to get the link for the `source` using the provider store, but we do not have the `provider` urls on the frontend. So, for Smithsonian museums, we only have the link to the individual museums (`source`) but not Smithsonian itself (`provider`)

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2731 by @obulat

## Description

This PR updates the metadata section of the single result page.

The metadata items match the updates styles: [Metadata component in Figma](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5171-16563&mode=design&t=blxAwawVa4wBfeT4-4), [Metadata in the audio page in Figma](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5645-29968&mode=design&t=wlOZdKnd7mwKmcQa-4)

The items don't wrap anymore, unless they don't fit. You can see that in the snapshot comparisons: previously, the long values would be split between 2-3 lines, and now they are all on one line.

The external links to the source, the provider and the album have an "external link" icon added to them.

The "source" link is updated to link to the source's landing page instead of the media's foreign landing URL.

## Testing Instructions

Check that the single result page metadata section matches the mockups. The code should make sense, too.
When you click on the links, correct analytics events should be fired, and only one per link (not external link click + some other event).

_Note:_ the (i) information popover explaining the difference between provider and source is not in scope of this PR

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/types/analytics.ts [hunk 1]
```diff
@@ -216,6 +216,18 @@ export type Events = {
     /** The permalink to the creator's profile */
     url: string
   }
+  /**
+   * Description: The user visits a source's link in the single result UI
```
Comment (obulat) [inline frontend/src/types/analytics.ts:220]: This event is for the external link to the source. Here, we can also get the id of the single result from which the link was clicked. However, for other source links (on the source collection page), there is no single media. Do we want to use the `id` as an optional prop, or is it not necessary?

File: frontend/src/types/analytics.ts [hunk 2]
```diff
@@ -216,6 +216,18 @@ export type Events = {
...
+   * Questions:
+   *   - Are source links clicked much? Does Openverse increase visibility
+   *     of included sources?
+   */
+  VISIT_SOURCE_LINK: {
+    /** The unique ID of the media */
+    id: string
```
Comment (obulat) [inline frontend/src/types/analytics.ts:227]:
```suggestion
    id?: string
```

**Review by fcoveram — COMMENTED** [review]
The change looks great.

### Focus ring

I just noticed that the UI's side areas crop the focus ring. See how subtle the component corners render parts of the ring.

Metadata component with glitchy pixels on all corners

I wonder if there is a site-wide solution that keeps the spacing values and sets the focus ring above all UI layers. Other parts of the interface face the same issue.

### Section layout

On the other hand, the new col-gap spacing in [this snapshot](https://github.com/WordPress/openverse/pull/2851/files#diff-639e5d75aab15327df7c26711e5fbca1c0d35087e3f8d803e888c869e83457aa) looks great, but [here](https://github.com/WordPress/openverse/pull/2851/files#diff-22550982884395bcd893fc6efdf85b4ff836adbd7f48dc652e0b7be40081671c) it looks unbalanced due to the album's name length.

The solution implemented works well, but to display a balanced layout we can try a max-width of `160px (w-20)` in the component, and place them in columns where items in second row follow the width of same item in first row. Like the quick mockups atttached below. Otherwise, item in second row will look off the grid, like [this snapshot](https://github.com/WordPress/openverse/pull/2851/files#diff-188bf15db0d9751637cc41b7cdf8b70d9bc39254352972d28d5848b62cc919ef).

Metada components in media information section. Guidelines are visible in the component borders

**Comment by fcoveram** [comment]
I forgot to add that on `xs / sm`, the category's label width (left column) could be the same as the longest label (Dimensions, as in the example attached below) to avoid the big gap between the two columns.

Metadata section on mobile, CSS values of category field are highlighted

File: frontend/src/pages/audio/_id/index.vue [hunk 3]
```diff
@@ -138,21 +138,3 @@ export default defineComponent({
   head: {},
 })
 </script>
```
Comment (obulat) [inline frontend/src/pages/audio/_id/index.vue:140]: These styles were not used: we don't have "audio-page" class anymore.

**Review by fcoveram — COMMENTED** [review]
The horizontal layout looks correct, but focus ring is still cropped by the content.

Not sure if the suggestion [I shared](https://user-images.githubusercontent.com/895819/263685555-985a3a35-7a5f-4bc3-b17d-9184d12ada77.png) above is possible, but I found [this example](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function) on MDN Web Docs that seems to address it.

**Comment by AetherUnbound** [comment]
image

Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

**Comment by fcoveram** [comment]
> Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

Yes. It's part of the design

**Comment by obulat** [comment]
@fcoveram, I had to add a small padding in the `xs` size to get the focus rings to show. Let me know what you think

**Comment by AetherUnbound** [comment]
Sorry to be clear:
image
It looks like "Dimensions" is aligned to the top of the box I've drawn, where as "1024 x 819 pixels" is aligned to the bottom. That's intended behavior?

**Comment by obulat** [comment]
> Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

Thank you for adding the snapshots, @AetherUnbound. The misalignment of items was caused by the padding I added to the links to make sure that they show the focus ring when you focus them with the keyboard (without the padding, they were getting cut off). I've also added the padding to the titles, so now the titles should be aligned with the items, as you can see in the `xs` snapshots in this PR.

**Review by fcoveram — DISMISSED** [review]
The change looks great. The padding is a good solution and is worth considering the same approach when implementing the tag component to reach the same layout spacing.

Regarding changes, and since the padding increases the gap between elements, I think we can change the following:
* On `xs` change the gap between rows from `24px` to `8px`.
* On `sm` and beyond
   * Change the gap between columns from `40px` to `32px`.
   * Change the padding-bottom of category field from `8px` to `4px`.

**Review by dhruvkb — COMMENTED** [review]
From the screenshots, it looks like the genre is not cased properly. What used to be 'Titlecase' is now all 'lowercase'.

**Review by AetherUnbound — APPROVED** [review]
LGTM, thanks for correcting the padding on the smaller sizes!

**Pull request merged**